tools/firmware.toit

```diff
@@ -60,26 +74,31 @@ main arguments/List:
   root_cmd.run arguments
 
 create_cmd -> cli.Command:
+  options := AR_ENTRY_FILE_MAP.map: | key/string value/string |
+    cli.OptionString key
+        --short_help="Set the $key part."
+        --type="file"
+        --required=(key == "firmware.bin")
   return cli.Command "create"
-      --options=[
-          cli.OptionString "binary"
-              --short_help="Set the binary input (e.g. firmware.bin)."
-              --type="file"
-              --required
-      ]
+      --options=options.values
       --run=:: create_envelope it
 
 create_envelope parsed/cli.Parsed -> none:
   output_path := parsed[OPTION_ENVELOPE]
-  input_path := parsed["binary"]
+  input_path := parsed["firmware.bin"]
 
   // TODO(kasper): Do some sanity checks on the
   // structure of this. Can we check that we don't
   // already have stuff appended to the DROM section?
-  binary_data := file.read_content input_path
-  binary := Esp32Binary binary_data
+  firmware_bin_data := file.read_content input_path
+  Esp32Binary firmware_bin_data
```

Comment:
Is this just to check that the data is correct?

Reply:
Yes. It is what I intend to extend with more validation.